flake.nix

```diff
@@ -34,9 +34,23 @@
     flake-utils.lib.eachDefaultSystem (
       system:
       let
+        wasm-bindgen-cli-overlay = final: prev:
+          let
+            cargoLock = builtins.fromTOML (builtins.readFile ./Cargo.lock);
+            wasmBindgenCargoVersions = builtins.map ({ version, ... }: version) (builtins.filter ({ name, ... }: name == "wasm-bindgen") cargoLock.package);
+            wasmBindgenVersion = assert builtins.length wasmBindgenCargoVersions == 1; builtins.elemAt wasmBindgenCargoVersions 0;
+          in
+          {
+            wasm-bindgen-cli = prev.wasm-bindgen-cli.override {
+              version = wasmBindgenVersion;
+              hash = "sha256-f/RK6s12ItqKJWJlA2WtOXtwX4Y0qa8bq/JHlLTAS3c=";
+              cargoHash = "sha256-3vxVI0BhNz/9m59b+P2YEIrwGwlp7K3pyPKt4VqQuHE=";
```

Comment:
Does hardcoding the hashes here pin this, regardless of what's in `Cargo.lock`? Presumably, when there's version drift, Nix will fail -- which will be caught by the CI, at the least -- when the hashes no longer match. Perhaps that's the best that can be done cleanly...

Reply:
Exactly, yes! When there is a version mismatch Nix will report an error that looks like this:

```fish
error: hash mismatch in fixed-output derivation '/nix/store/zg1xxw53cm7fny593n7fy8ymns5bk78g-wasm-bindgen-cli-0.2.91.tar.gz.drv':
         specified: sha256-OdK/fVWOpbMBsl37pSWawPqpk5sePqtu1lx1UM+7c9Q=
            got:    sha256-f/RK6s12ItqKJWJlA2WtOXtwX4Y0qa8bq/JHlLTAS3c=
```

Which is trivial to resolve.